playwright/tests/pageobjects/CodeEditor.ts

```diff
@@ -66,6 +75,32 @@ class CodeEditor {
   }
 }
 
+export class Browser {
+  constructor(readonly page: Page) {}
+
+  async openTab(name: string) {
+    const tab = this.page.locator('button.tabs-trigger', { hasText: name });
+    tab.click();
+    await expect(tab).toHaveAttribute('aria-selected', 'true');
+  }
+
+  async search(lookup: string) {
+    const search = this.dialog().getByRole('textbox');
+    await search.click();
+    await search.focus();
+    await search.fill('');
+    await search.pressSequentially(lookup);
+  }
+
+  dialog() {
+    return this.page.getByRole('dialog');
+  }

```

Comment:
if you don't like the () in the call you can also use the get function syntax (just as hint)
```ts
get dialog() {
  return this.page.getByRole('dialog');
}

//then you can instead of
browser.dialog().locator...
//you can do
browser.dialog.locator
```